go/vt/vtgate/planbuilder/operators/query_planning.go

```diff
@@ -107,6 +107,13 @@ func runRewriters(ctx *plancontext.PlanningContext, root Operator) Operator {
 		}
 	}
 
+	if pbm, ok := root.(*PercentBasedMirror); ok {
+		root = pbm.Clone([]Operator{
```

Comment:
Why the need to clone, why not just set inputs?

Reply:
no reason, i didn't think to use set inputs.